### PR TITLE
Migrated UserInputHelper into Fake.Core.UserInput

### PR DESCRIPTION
**Pull request by nelak**

### CODE REVIEW

File: src/app/Fake.Core.UserInput/UserInput.fs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+open System
+
+/// Helpers for capturing user input
+///
+/// ## Sample
+///
+///     UserInput.getUserInput prompt
+[<RequireQualifiedAccess>]
+module UserInput =
+let internal erasePreviousChar () =
```
Comment (matthid) [inline src/app/Fake.Core.UserInput/UserInput.fs]: The code if the module needs to be intended to the right in order to get rid of the warning:

C:\projects\fake-6w516\src\app\Fake.Core.UserInput\UserInput.fs(11,1): warning FS0058: Possible incorrect indentation: this token is offside of context started at position (10:1). Try indenting this token further or using standard formatting conventions. [C:\projects\fake-6w516\src\app\Fake.Core.UserInput\Fake.Core.UserInput.fsproj]

Basically just intend from line 11 to line 63

File: src/app/Fake.Core.UserInput/UserInput.fs [hunk 2]
```diff
@@ -0,0 +1,63 @@
+namespace Fake.Core
+open System
+
+/// Helpers for capturing user input
+///
+/// ## Sample
+///
+///     UserInput.getUserInput prompt
+[<RequireQualifiedAccess>]
+module UserInput =
```
Comment (matthid) [inline src/app/Fake.Core.UserInput/UserInput.fs]: This module is copied from an existing one from `FakeLib`, correct? Can we change the obsolete warnings of the existing module in `FakeLib` to point to this one?

Comment (matthid) [reply src/app/Fake.Core.UserInput/UserInput.fs]: One extension to that: We also need to add this new module to `FakeLib` in order to enable the migration path.

Comment (nelak) [reply src/app/Fake.Core.UserInput/UserInput.fs]: This should be addressed by linking the file in FakeLib which referenced by the Legacy-FAKE solution right?

**Comment by nelak** [comment]
I believe this should be ready now, let me know if further adjustments are needed

**Comment by nelak** [comment]
Added the reference in `templates.cs` based on the links generated found in `docs/apidocs`

**Comment by matthid** [comment]
Yes exactly, thanks!
